samcli/lib/build/build_strategy.py

```diff
@@ -125,14 +125,18 @@ def build_single_function_definition(self, build_definition: FunctionBuildDefini
         function_build_results[single_function_name] = result
 
         # copy results to other functions
-        if build_definition.packagetype == ZIP:
-            for function in build_definition.functions:
-                if function.name is not single_function_name:
+        for function in build_definition.functions:
+            if function.name != single_function_name:
+                if build_definition.packagetype == ZIP:
+                    # for zip function we need to copy over the artifacts
```

Comment:
Why we need to change the order here? This will check the `if` statement for every function in the build definition (no matter it is `ZIP` or `IMAGE`), but the previous order will have 2 separate flow for `IMAGE` and `ZIP` from the beginning.

Reply:
right, good catch! I optimized for more less code. This indeed introduced more checks. Reverted